operator/src/main/java/oracle/kubernetes/operator/helpers/PodHelper.java

```diff
@@ -505,6 +538,120 @@ protected String getClusterName() {
       addStartupEnvVars(vars);
       return vars;
     }
+
+    abstract class ExporterContext {
+      int getWebLogicRestPort() {
+        return selectPortByProtocolName().orElse(selectPortFromList());
+      }
+
+      private Optional<Integer> selectPortByProtocolName() {
+        return getContainerPorts().stream()
+              .filter(p -> Objects.equals(scan.getAdminProtocolChannelName(), p.getName()))
+              .findFirst()
+              .map(V1ContainerPort::getContainerPort);
+      }
+
+      private int selectPortFromList() {
+        return Stream.of(getAdminPort(), getListenPort(), getSslListenPort())
+              .filter(Objects::nonNull)
+              .findFirst()
+              .orElseThrow(() -> new RuntimeException("No ports defined for this server"));
+      }
+
+      boolean isWebLogicSecure() {
+        return !Objects.equals(getWebLogicRestPort(), getListenPort());
+      }
+
+      abstract boolean isEnabled();
+
+      abstract int getPort();
+
+      abstract String getBasePath();
+
+      abstract void addContainer(List<V1Container> containers);
+    }
+
+    class WebAppExporterContext extends ExporterContext {
+
+      @Override
+      boolean isEnabled() {
+        return getListenPort() != null;
+      }
+
+      @Override
+      int getPort() {
+        return getListenPort();
```

Comment:
Does this need to be SSL port if the admin channel is enabled or SSL is enabled?

Reply:
It gets complicated, as Prometheus is currently assuming plainport access to the exporter. But this is only for the web app case, which we are hoping to deprecate within the operator.